README.md

```diff
@@ -1,9 +1,205 @@
 [<img alt="libfabric master branch Travis CI status" src="https://travis-ci.org/ofiwg/libfabric.svg?branch=master"/>](https://travis-ci.org/ofiwg/libfabric)
 [<img alt="libfabric Coverity scan suild status" src="https://scan.coverity.com/projects/4274/badge.svg"/>](https://scan.coverity.com/projects/4274)
 
-libfabric
-=========
+# libfabric
 
-The Open Fabrics Interfaces (OFI) is a framework focused on exporting fabric communication services to applications.
+The Open Fabrics Interfaces (OFI) is a framework focused on exporting fabric
+communication services to applications.
 
-See [the OFI web site](http://ofiwg.github.io/libfabric/) for more details, including a description and overview of the project, and detailed documentation of the libfabric APIs.
+See [the OFI web site](http://libfabric.org) for more details, including a
+description and overview of the project, and detailed documentation of the
+libfabric APIs.
+
+## Building libfabric
+
+If you are building Libfabric from a developer Git clone, you must first run
+the `autogen.sh` script. This will invoke the GNU Autotools to bootstrap
+Libfabric's configuration and build mechanisms. If you are building Libfabric
+from an official distribution tarball from libfabric.org, there is no need to
+run `autogen.sh`; Libfabric distribution tarballs are already bootstrapped for
+you.
+
+Libfabric currently supports GNU/Linux, Free BSD, and OS X.
+
+### Dependencies
+
+- `libpthread`
+- `libdl`
```

Comment:
NM... the very next sentence says that provider dependencies are discussed below... :-)